### DISCUSSION TITLE
Welcome to HermitsHeresy Discussions!

### DISCUSSION THREAD

**Discussion by default-kramer** [default-kramer/HermitsHeresy#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Sapphire645** [comment]
I'm not sure how this works... but as info sources I have the block icons and overview tiles here https://github.com/Sapphire645/DQB2ChunkEditorPlus/tree/main/src/Images 

and an incomplete ID list here https://github.com/Sapphire645/DQB2ChunkEditorPlus/blob/main/src/Data/Tiles.json

and here is the overview in game pictures from dumping all ids in order https://www.tumblr.com/sapphire-rb/751291945137078272/all-of-dqb2s-blocks-with-their-ids

Hopefully this is is useful. I will update the github repository icons and tile info in the future. It's honestly a little bit (very) tedious work so the less times it's repeated by different people the better.

**Reply by default-kramer** [reply]
Thanks! It is tedious and I was planning on filling in the gaps in my blocklist from other sources before I release v0.1 of Hermit's Heresy. This will be a good source.

**Reply by Sapphire645** [reply]
Ah! Also, all of the blocks placed down like that on the images was generated with this (very bad vey awful) python code. If you find the first block on a Y layer of the chunk and paste one batch of the text generated you'll get the blocks placed like that in-game. So for block testing and documentation it might be useful?

You just place 8 blocks at the beggining of 8 chunks, find them later in hex editor (with the decoded file), then overwrite that block id with the text chunk generated with the script.

The last 2 are for clearing out the layer with air and setting down grass on that layer.

Then you can go for each one and try to break them, see what they give, and all of that.
[IdGenerationCode.txt](https://github.com/default-kramer/HermitsHeresy/files/15448659/IdGenerationCode.txt)

I saw the item documentation. Unfortunately I don't have time right now for research, but I'm following it closely.

**Comment by Sapphire645** [comment]
Hey, I've paused looking at the item stuff for a while, and I've been documenting the unused stuff ofthe game instead. If you get items working and find some unused ones could you take pics of them for documentation? (Only if you run into some, it's uncharted territory after all!)

The debug build has an item viewer that lists the IDs, but I think it differs from the actual IDs from the normal release. There's also an option that makes IDs appear on top of the items, and that can be used to identify the ID of items quickly, (if the ID is the same in the normal release)

**Reply by default-kramer** [reply]
I'll keep an eye out. I'm not getting much hacking done right now, but I hope to get back into it pretty soon.

**Comment by Sapphire645** [comment]
I'm working on listing out the chunks for the maps, the images might be useful for you?
<img src="https://github.com/user-attachments/assets/ae783d36-a096-498d-8ee1-602268a17ad2">
<img src="https://github.com/user-attachments/assets/3fe62a20-0552-4127-849d-c1428334f0af" width="500"> <img src="https://github.com/user-attachments/assets/34b415ae-4906-4883-a976-28b61782af2a" width="500">
Only have IoA right now (the one you already have oops) but I'm doing the other islands too. 

I'm always an advocate for "if someone's already done the work then there's no need to do it again" so if anything I do is ever useful for you then I'm happy. (specially if its tedious work like the block documentation or chunk mapping)

If we figure out how minimaps work we could even do a little program that prints out the island maps if given the CMNDAT? That way anyone can have high quality island renders and use them for this proyect or just for sharing online!

**Reply by default-kramer** [reply]
Yep I'm starting to imagine the usefulness of being able to read the minimap data. Let me know if you get anywhere with it. I may make an attempt myself after I get a few other things done.

**Reply by Sapphire645** [reply]
Do you know how to take a value and assing it to a tile, then print that tile into an image that can be saved?
